nucypher/blockchain/eth/agents.py

```diff
@@ -185,6 +185,8 @@ def approve_transfer(self,
                          transacting_power: TransactingPower
                          ) -> TxReceipt:
         """Approve the spender address to transfer an amount of tokens on behalf of the sender address"""
+        self._validate_zero_allowance(spender_address, transacting_power)
```

Comment:
if amount is zero and allowance is not then `_validate_zero_allowance` will fail (but it should not)